### PR TITLE
Use TestUtil.problemsService() consistently in functional tests

### PR DESCRIPTION
**Pull request by donat**
Fixes https://github.com/gradle/bt-ide-experience/issues/33

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test QFL

File: subprojects/core/src/testFixtures/groovy/org/gradle/util/TestUtil.groovy [hunk 1]
```diff
@@ -365,3 +366,68 @@ class TestUtil {
...
+
+    @Override
+    AdditionalDataBuilderFactory getAdditionalDataBuilderFactory() {
+        delegate.additionalDataBuilderFactory
+    }
+
+    void assertProblemEmittedOnce(Object expectedProblem) {
+        assert summarizer.emitted.size() == 1
+        def problem = summarizer.emitted[0]
+        assert expectedProblem instanceof Closure ? expectedProblem.call(problem) : expectedProblem instanceof Wildcard ? true : expectedProblem == problem
```
Comment (hegyibalint) [inline subprojects/core/src/testFixtures/groovy/org/gradle/util/TestUtil.groovy:403]: :x: this is really hard to read. Could you break this down into an if-else structure? I think doing that wouldn't break the `assert` statement.

**Review by hegyibalint — CHANGES_REQUESTED** [review]

File: subprojects/core/src/testFixtures/groovy/org/gradle/util/TestUtil.groovy (hunk 1 above)
Comment (donat) [reply subprojects/core/src/testFixtures/groovy/org/gradle/util/TestUtil.groovy]: Fixed, please take another look.

File: subprojects/core/src/testFixtures/groovy/org/gradle/util/TestUtil.groovy [hunk 2]
```diff
@@ -365,3 +366,74 @@ class TestUtil {
...
+
+    void assertProblemEmittedOnce(Object expectedProblem) {
+        assert summarizer.emitted.size() == 1
+        def actualProblem = summarizer.emitted[0]
+        if (expectedProblem instanceof Closure) {
+            assert expectedProblem.call(actualProblem)
+        } else if (expectedProblem instanceof Problem) {
+            assert expectedProblem == actualProblem
+        } else {
+            assert expectedProblem instanceof Wildcard
```
Comment (hegyibalint) [inline subprojects/core/src/testFixtures/groovy/org/gradle/util/TestUtil.groovy:408]: ℹ️ TIL how `_` works in Spock, it becomes a wildcard! Interesting.

**Review by hegyibalint — APPROVED** [review]
Thanks for the changes, let's merge this!

**Comment by donat** [comment]
@bot-gradle merge

**Comment by donat** [comment]
@bot-gradle merge

**Pull request merged**